src/psyclone/tests/psyir/nodes/array_mixin_test.py

```diff
@@ -40,36 +40,116 @@
+    and use an ArrayReference node to test the abstract ArrayMixin
+    class.
 
-def test_get_outer_range_index():
-    '''Check that the get_outer_range_index method returns the outermost index
-    of the children list that is a range. Use ArrayReference and
-    ArrayOfStructuresReference as concrete implementations of ArrayMixins.
     '''
-    symbol = DataSymbol("my_symbol", ArrayType(INTEGER_TYPE, [10, 10, 10]))
-    array = ArrayReference.create(symbol, [Range(), Range(), Range()])
-    assert array.get_outer_range_index() == 2
+    one = Literal("1", INTEGER_TYPE)
+    array = DataSymbol("my_symbol", ArrayType(INTEGER_TYPE, [10]))
+    array2 = DataSymbol("my_symbol2", ArrayType(INTEGER_TYPE, [10]))
+    scalar = DataSymbol("tmp", INTEGER_TYPE)
+    ubound = BinaryOperation.create(
+        BinaryOperation.Operator.UBOUND, Reference(array), one)
+    my_range = Range.create(one.copy(), ubound)
+    array_ref = ArrayReference.create(array, [my_range])
+    array2_ref = ArrayReference.create(array2, [my_range.copy()])
+    # not a binary operation
+    oper = None
+    assert not array_ref._is_bound_op(
+        oper, BinaryOperation.Operator.UBOUND, one)
+    # not a match with the binary operator
+    oper = BinaryOperation.create(
+        BinaryOperation.Operator.LBOUND, array_ref, one.copy())
+    assert not array_ref._is_bound_op(oper, BinaryOperation.Operator.UBOUND, 0)
+    # 1st dimension of the bound is not the same array
+    oper = BinaryOperation.create(
+        BinaryOperation.Operator.UBOUND, array2_ref, one.copy())
+    assert not array_ref._is_bound_op(oper, BinaryOperation.Operator.UBOUND, 0)
+    # 2nd dimension of the bound not a literal
+    oper = BinaryOperation.create(
+        BinaryOperation.Operator.UBOUND, array_ref.copy(), Reference(scalar))
+    assert not array_ref._is_bound_op(oper, BinaryOperation.Operator.UBOUND, 0)
+    # 2nd dimension of the bound not an integer literal
+    oper = BinaryOperation.create(
+        BinaryOperation.Operator.UBOUND, array_ref.copy(),
+        Literal("1.0", REAL_TYPE))
+    assert not array_ref._is_bound_op(oper, BinaryOperation.Operator.UBOUND, 0)
+    # 2nd dimension of the bound not the expected index
+    oper = BinaryOperation.create(
+        BinaryOperation.Operator.UBOUND, array_ref.copy(),
+        Literal("2", INTEGER_TYPE))
+    assert not array_ref._is_bound_op(oper, BinaryOperation.Operator.UBOUND, 0)
+    # OK
+    oper = BinaryOperation.create(
+        BinaryOperation.Operator.UBOUND, array_ref.copy(), one.copy())
+    assert array_ref._is_bound_op(oper, BinaryOperation.Operator.UBOUND, 0)
 
-    symbol = DataSymbol("my_symbol", DeferredType())
-    aos = ArrayOfStructuresReference.create(
-        symbol, [Range(), Range(), Range()], ["nx"])
-    assert aos.get_outer_range_index() == 3  # +1 for the member child
 
+# _get_symbol_declaration
+
+def test_get_symbol_declaration(fortran_reader):
```

Comment:
Hopefully this whole test can be removed.

Reply:
Yes, the tests have been reworked given the change to the implementation of `is_lower()` and `is_upper()`.